### ISSUE TITLE
[HOLD for payment 2023-11-29] [$500] Status - Different chat report opens when clicking on the back button in Status modal

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#30509]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.3.92-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team
**Slack conversation:** @

## Action Performed:
Precondition: User has set a status.

1. Go to staging.new.expensify.com
2. Open different chat reports.
3. Click on the status on the top left.
4. Click on the back button.

## Expected Result:
The current chat report remains open.

## Actual Result:
A different chat report opens.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/115492554/fa4a4085-ec8e-45cf-8c4b-47eda72429b7

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f903eb3ac10fe198</li>
        <li>Upwork Job ID: 1717888354588504064</li>
        <li>Last Price Increase: 2023-10-27</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Reviewer | 27537929</li>
<li>dukenv0307 | Contributor | 27537933</li>
</ul></li>
    </ul>
</details>

**Comment by s-alves10** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When go to status page after navigating several reports, back button navigates to the report other than the current report

### What is the root cause of that problem?

When we navigate to status page, we set `shouldPopAllStateOnUP` to true and go to the page
https://github.com/Expensify/App/blob/1fc2bbc030603b4768c522cec2a48089cae03da3/src/pages/home/sidebar/AvatarWithOptionalStatus.js#L36
And when we go back, we call `navigate.goBack` with `shouldPopToTop = true`
https://github.com/Expensify/App/blob/1fc2bbc030603b4768c522cec2a48089cae03da3/src/pages/settings/Profile/CustomStatus/StatusPage.js#L53

This makes the route stack pop to the first page, which is home page. 
https://github.com/Expensify/App/blob/1fc2bbc030603b4768c522cec2a48089cae03da3/src/libs/Navigation/Navigation.js#L107-L113

For wide screen devices, we navigate to the last accessed report when navigating to home page
https://github.com/Expensify/App/blob/ee5e6de5e8c34f8f60d6d78f4b83aa159c1e9dec/src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js#L86

We determine the last access report by its lastReadTime.
https://github.com/Expensify/App/blob/ee5e6de5e8c34f8f60d6d78f4b83aa159c1e9dec/src/libs/ReportUtils.js#L553

The problem is here. When we call `OpenReport` API, `lastReadTime` isn't updated


### What changes do you think we should make in order to solve the problem?


<details>
    <summary>Result</summary>



</details>

### What alternative solutions did you explore? (Optional)

**Comment by CortneyOfstad** [comment]
@shubham1206agra thoughts on the proposals above? Thanks!

**Comment by shubham1206agra** [comment]
@dukenv0307's [proposal](https://github.com/Expensify/App/issues/30509#issuecomment-1783703740) works correctly.
But @s-alves10 comment about `lastReadTime` not being updated seems correct, too.
We can go with @dukenv0307, but I think we should discuss the backend call for the `OpenReport` API with an engineer.
🎀 👀 🎀 C+ reviewed

**Comment by s-alves10** [comment]
I suggest to try with backend updates updating `lastReadTime` first. Obviously this is a bug, I think. If it doesn't solve the issue. we can go with @dukenv0307 's proposal

**Comment by shubham1206agra** [comment]
Ok let's go with @dukenv0307 proposal then

**Comment by s-alves10** [comment]
> Even after that BE issue is fixed, if you open the Status page, then in another device you read another report, then coming back to this device and click back on Status modal, it will go to that other report rather than the report in the background of the Status page.

@dukenv0307 

Maybe you're right, but from my research, I don't think so

By the way, don't you think not updating `lastReadTime` is a bug? I am sure we need to update the `lastReadTime` when we call `OpenReport` API call. That's why I suggested to fix the backend first.

Anyway, these are just my personal thoughts.

**Comment by shubham1206agra** [comment]
@s-alves10 Please don't worry about this. Let me ask this in Slack.

**Comment by shubham1206agra** [comment]
> Is there any specific reason we do not update lastReadTime  when opening a report using OpenReport  API call?

Yes, because with OpenReport we're just fetching reports for user from the server. The report is read when we actually render them on the screen. These are two closely related but different actions and hence we have different API calls.
More information is in the description of this issue https://github.com/Expensify/App/issues/23171

This is the answer I got @s-alves10

**Comment by shubham1206agra** [comment]
@Beamanator Friendly bump

**Comment by Beamanator** [comment]
@shubham1206agra thanks for bringing that convo to slack, can you please link it here so it's easy access? If you're having trouble getting that separate issue to reopen, we may also want to bring that up in slack (if you haven't already)

So @shubham1206agra you're still advocating for going with @dukenv0307 's proposal, right? Anything you need from me other than a review as well?

**Comment by shubham1206agra** [comment]
@Beamanator You can proceed with @dukenv0307 right now

I do have seperate issue for this too which are open. I will start a discussion there only (maybe in slack too)

**Comment by Beamanator** [comment]
Thanks @shubham1206agra !

**Comment by shubham1206agra** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

* [x]  [@shubham1206agra] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/24414
* [x]  [@shubham1206agra] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/24414/files#r1404397736
* [x]  [@shubham1206agra] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: https://expensify.slack.com/archives/C02NK2DQWUX/p1698756003834449 (This old discussion is in #contributor-plus channel)
* [x]  [@shubham1206agra] Determine if we should create a regression test for this bug. Yes
* [x]  [@shubham1206agra] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
1. Open the App with an account that has a status set already.
2. Open different chat reports.
3. Click on the status on the top left.
4. Click on the back button.

**Comment by CortneyOfstad** [comment]
@shubham1206agra @dukenv0307 — both of you have been paid in NewDot and regression test was created [here](https://github.com/Expensify/Expensify/issues/341483) — closing!
